### PR TITLE
Added support for AMQP certificates, refactored Cert handling into a common connector-utils module

### PR DESCRIPTION
**Pull request by dhirajsb**

### CODE REVIEW

**Comment by dhirajsb** [comment]
Fixes #1225

File: app/connectors/connector-utils/pom.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    <version>1.3-SNAPSHOT</version>
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>connector-utils</artifactId>
+  <name>Connector :: Utils</name>
+  <packaging>jar</packaging>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (zregvart) [inline app/connectors/connector-utils/pom.xml]: I think there is no need for this, it should be inherited from syndesis-parent

File: app/connectors/connector-utils/pom.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.6.0</version>
+        <configuration>
+          <source>1.8</source>
+          <target>1.8</target>
+        </configuration>
+      </plugin>
```
Comment (zregvart) [inline app/connectors/connector-utils/pom.xml]: I think there is no need for this, it should be inherited from syndesis-parent

File: app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.stream.Collectors;
+
+/**
+ * Helper class to create KeyStores from Certificates.
+ *
+ * @author dhirajsb
+ */
+public class KeyStoreHelper {
+
+    private static final SecureRandom SECURE_RANDOM = new SecureRandom();
```
Comment (zregvart) [inline app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java]: Could use [SecureRandom.getInstanceStrong()](https://docs.oracle.com/javase/8/docs/api/java/security/SecureRandom.html#getInstanceStrong--), though if it's used for generating passwords, [ThreadLocalRandom.current()](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadLocalRandom.html#current--) ought to be enough random.

File: app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        return password;
+
+    }
+
+    public KeyStoreHelper store() {
+        try {
+
+            KeyStore keyStore = CertificateUtil.createKeyStore(certificate, alias);
+
+            tempFile = Files.createTempFile(alias, ".ks", PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rwx------")));
```
Comment (zregvart) [inline app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java]: We might not need the execute permission bit

File: app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            password = generatePassword();
+
+            try (OutputStream stream = new FileOutputStream(tempFile.toFile())) {
+                keyStore.store(stream, password.toCharArray());
+            }
+
+        } catch (KeyStoreException | IOException | NoSuchAlgorithmException | CertificateException e) {
```
Comment (zregvart) [inline app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java]: Could be `catch (GeneralSecurityException | IOException e)`

File: app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            throw new IllegalArgumentException(String.format("Error creating key store %s: %s", alias, e.getMessage()), e);
+        }
+
+        return this;
+    }
+
+    private static String generatePassword() {
+        return SECURE_RANDOM.ints(16, 'A', 'Z' + 1).mapToObj(i -> String.valueOf((char)i)).collect(Collectors.joining());
```
Comment (zregvart) [inline app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java]: Could be:

```java
int[] codepoints = SECURE_RANDOM.ints(16, 'A', 'Z' + 1).toArray();

return new String(codepoints, 0, 16);
```

see https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#String-int:A-int-int-

File: app/connectors/connector-utils/src/test/java/io/syndesis/connector/utils/CertificateUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            sb.append("\nProvider : " + p[i].toString() + "\n");
+            Set s = p[i].keySet();
+            Object[] o = s.toArray();
+            Arrays.sort(o);
+            for (int j = 1; j < o.length; j++) {
+                sb.append(o[j].toString() + ", ");
+            }
+        }
+        System.out.println(sb.toString());
+*/
```
Comment (zregvart) [inline app/connectors/connector-utils/src/test/java/io/syndesis/connector/utils/CertificateUtilTest.java]: can we remove this commented out code?

File: app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java (hunk 3 above)
Comment (dhirajsb) [reply app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java]: I used to think that `getInstanceStrong()` was preferable too, until I read [this](https://tersesystems.com/blog/2015/12/17/the-right-way-to-use-securerandom/).

File: app/connectors/connector-utils/src/test/java/io/syndesis/connector/utils/CertificateUtilTest.java (hunk 7 above)
Comment (dhirajsb) [reply app/connectors/connector-utils/src/test/java/io/syndesis/connector/utils/CertificateUtilTest.java]: yes, we can!

File: app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java (hunk 3 above)
Comment (dhirajsb) [reply app/connectors/connector-utils/src/main/java/io/syndesis/connector/utils/KeyStoreHelper.java]: BTW, we also do [this](https://github.com/fabric8io-images/s2i/blob/39f99bf1755809069576233e914f3bc0eb7fd8d3/java/images/rhel/Dockerfile#L29).

**Pull request merged**